update gobgp to 2.19.0

murali-reddy opened the pull request.

aauren commented:
Thanks for starting this off @murali-reddy! Let me know when it's in a place for review.

murali-reddy commented:
This PR is ready for review. I have done significant testing to cover below functional areas

- basic pod CIDR's of nodes are exchanges between the iBGP nodes
- routes to pod CIDR's are installed to local FIB based on the updates to the gobgp RIB
- service VIP's are advertised to external peers
- appropriate BGP import and export policies as required by kube-router are configured
- all prefix sets and neighbor sets used by BGP policies are populated and updated as the nodes, services gets addedd/deleted
- route reflector configuration

Also all unit tests are modified as per the gobgp 2.* API's and are passing.

mrueg commented on pkg/controllers/routing/bgp_peers.go:
```diff
@@ -144,22 +141,18 @@ func (nrc *NetworkRoutingController) syncInternalPeers() {
 			}
 		}
 
 		// TODO: check if a node is alredy added as nieighbour in a better way than add and catch error
```
Is that TODO still required? (also there's a small typo in neighbor)

mrueg commented on pkg/controllers/routing/bgp_policies.go:
```diff
@@ -1,141 +1,345 @@
+	return externalBGPPeerCIDRs, nil
+}
+
+// a slice of all peers is used as a match condition for reject statement of servicevipsdefinedset import polcy
```
typo: policy

mrueg commented on pkg/controllers/routing/network_routes_controller.go:
```diff
@@ -714,18 +750,14 @@ func (nrc *NetworkRoutingController) startBgpServer() error {
 	}
 
-	nrc.bgpServer = gobgp.NewBgpServer()
+	//nrc.bgpServer = gobgp.NewBgpServer()
```
that can be removed?

murali-reddy commented:
For a referance the exposed consumable API from gobgp are present in

https://github.com/osrg/gobgp/blob/master/api/gobgp.pb.go
https://github.com/osrg/gobgp/blob/master/api/gobgp.proto

murali-reddy commented:
All the unit tests and lint errors fixed. https://travis-ci.org/github/cloudnativelabs/kube-router/builds/723788806  Addresses review comments as well.
